### PR TITLE
Create Observability AI Assistant docs

### PR DESCRIPTION
**Pull request by mdbirnstiehl**
This PR closes [Issue 3034](https://github.com/elastic/observability-docs/issues/3034).

I've added technical preview documentation for the Observability AI Assistant.

You can see a preview of the documentation [here](https://observability-docs_3058.docs-preview.app.elstc.co/guide/en/observability/master/obs-ai-assistant.html)

### CODE REVIEW

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+You can find Observability AI Assistant prompts throughout {observability}:
+
+[role="screenshot"]
+image::images/obs-assistant.gif[Observability AI assistant preview, 90%]
+
+[IMPORTANT]
+====
+Observability AI Assistant is in technical preview, and its capabilities are still developing. Users should leverage it sensibly as the reliability of its responses might vary. Your insights, patience, and feedback help us calibrate this feature for optimal use. Always cross-verify any returned advice for accurate threat detection and response, insights, and query generation.
+
+Also, the data you provide to the Security Assistant is _not_ anonymized, and is stored and processed by the third-party AI provider. This includes any data used in conversations for analysis or context, such as alert or event data, detection rule configurations, and queries. Therefore, be careful about sharing any confidential or sensitive details while using this feature.
```
Comment (dedemorton) [inline docs/en/observability/observability-ai-assistant.asciidoc:17]: Should this say "Security Assistant"?

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+* An account with a third-party generative AI provider, which the Observability AI Assistant uses to generate responses. Supported providers are OpenAI (`gpt-4` model) and Azure OpenAI Service (`gpt-4` model).
+
+[discrete]
+[[obs-ai-set-up]]
+= Set up the Observability AI Assistant
+
+Complete the following steps to use the Observability AI Assistant:
+
+. Create an API key with your AI provider to authenticate requests from the Security Assistant. You'll use this in the next step. Refer to the provider's documentation for generating API keys:
```
Comment (dedemorton) [inline docs/en/observability/observability-ai-assistant.asciidoc:36]: Same comment about security assistant...is this really what it's called?

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+* https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference[Azure OpenAI Service]
+
+. Update {kib}'s configuration settings to include the Observability AI Assistant feature flag and additional information about your AI set up.
++
+.. Navigate to the {kib} configuration settings according to your deployment:
++
+** *Self-managed (on-premises) deployments*: Add the configuration to the `kibana.yml` file, which is used to {kibana-ref}/settings.html[configure {kib}].
+** *{ecloud} deployments*: Use the YAML editor in the {ecloud} console to add the configuration to {cloud}/ec-manage-kibana-settings.html[{kib} user settings].
++
+.. Add one of the following configurations that corresponds the AI service you're using:
```
Comment (dedemorton) [inline docs/en/observability/observability-ai-assistant.asciidoc:48]:
```suggestion
.. Add one of the following configurations that corresponds to the AI service you're using:
```

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+You can find Observability AI Assistant prompts throughout {observability} that provide the following information:
+
+- *Universal Profiling* – explains the most expensive processes and functions in your fleet and provides optimization suggestions.
+- *Application performance monitoring (APM)* – explains APM errors and provides remediation suggestions.
+- *Infrastructure Observability* – explains host process and provides suggestions for optimizing CPU usage.
+- *Logs* – explains log messages and generates search patterns to find similar issues.
+- *Alerting* – provides possible log spike causes and remediation suggestions.
+
+
+For example, in the logs details, you'll see prompts for *What's this message?* and *How do I find similar log messages*:
```
Comment (dedemorton) [inline docs/en/observability/observability-ai-assistant.asciidoc:93]: Is "logs details" right here? "Log details" sounds better unless there's some precedent for making logs plural.

**Review by dedemorton — COMMENTED** [review]
Just some drive-by comments because I was curious about this new feature and wanted to learn more.

**Review by dedemorton — APPROVED** [review]
lgtm

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 5]
```diff
@@ -0,0 +1,101 @@
+[[obs-ai-assistant]]
+= Observability AI Assistant
+
+preview::[]
+
+The Observability AI Assistant is a large language model (LLM) integration that helps Elastic Observability users by adding context, explaining errors and messages, and suggesting remediation in the {observability} interface. A connector for OpenAI or Azure OpenAI Service powers the Observability AI Assistant.
```
Comment (SeanHeelan) [inline docs/en/observability/observability-ai-assistant.asciidoc:6]:
```suggestion
The Observability AI Assistant is a large language model (LLM) integration that helps Elastic Observability users by adding context, explaining errors and messages, and suggesting remediation in the {observability} interface. The Observability AI Assistant can be connected to either the OpenAI or Microsoft Azure LLM services. 
```

cc @dgieselaar Does this wording make more sense, as we're not using the connector framework?

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+You can find Observability AI Assistant prompts throughout {observability}:
+
+[role="screenshot"]
+image::images/obs-assistant.gif[Observability AI assistant preview, 90%]
+
+[IMPORTANT]
+====
+Observability AI Assistant is in technical preview, and its capabilities are still developing. Users should leverage it sensibly as the reliability of its responses might vary. Your insights, patience, and feedback help us calibrate this feature for optimal use. Always cross-verify any returned advice for accurate threat detection and response, insights, and query generation.
+
+Also, the data you provide to the Observability AI assistant is _not_ anonymized, and is stored and processed by the third-party AI provider. This includes any data used in conversations for analysis or context, such as alert or event data, detection rule configurations, and queries. Therefore, be careful about sharing any confidential or sensitive details while using this feature.
```
Comment (SeanHeelan) [inline docs/en/observability/observability-ai-assistant.asciidoc:17]: Just a comment on this: we also receive data that is submitted in prompts to OpenAI. I am awaiting feedback from @cconboy, who is in turn talking to legal, on what (if any) changes to this wording we need to make. @mdbirnstiehl I'll let you know when I hear anything on this front.m

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+[discrete]
+[[obs-ai-requirements]]
+= Requirements
+
+You need following to use the Observability AI Assistant:
+
+* {stack} version 8.9 and later.
+
+* An account with a third-party generative AI provider, which the Observability AI Assistant uses to generate responses. Supported providers are OpenAI (`gpt-4` model) and Azure OpenAI Service (`gpt-4` model).
```
Comment (SeanHeelan) [inline docs/en/observability/observability-ai-assistant.asciidoc:28]:
```suggestion
* An account with a third-party generative AI provider, which the Observability AI Assistant uses to generate responses. Supported providers are OpenAI and Azure OpenAI Service. You may use either the `gpt-3.5-turbo` or `gpt-4` models. 
```

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+** *OpenAI*:
++
+In the following example, `xpack.observability.aiAssistant.provider.openAI.apiKey` is the API key you generated in the first step.
++
+[source,yaml]
+----
+xpack.observability.aiAssistant.enabled: true
+#
+xpack.observability.aiAssistant.provider.openAI.apiKey: xxxx
+#
```
Comment (SeanHeelan) [inline docs/en/observability/observability-ai-assistant.asciidoc:59]:
```suggestion
xpack.observability.aiAssistant.provider.openAI.apiKey: xxxx
```

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
++
+** *OpenAI*:
++
+In the following example, `xpack.observability.aiAssistant.provider.openAI.apiKey` is the API key you generated in the first step.
++
+[source,yaml]
+----
+xpack.observability.aiAssistant.enabled: true
+#
+xpack.observability.aiAssistant.provider.openAI.apiKey: xxxx
```
Comment (SeanHeelan) [inline docs/en/observability/observability-ai-assistant.asciidoc:58]:
```suggestion
xpack.observability.aiAssistant.provider.openAI.apiKey: <insert API key>
```

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+# 
+xpack.observability.aiAssistant.provider.openAI.model: gpt-4
```
Comment (SeanHeelan) [inline docs/en/observability/observability-ai-assistant.asciidoc:60]:
```suggestion
xpack.observability.aiAssistant.provider.openAI.model: <insert model name, e.g. gpt-4>
```

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+. If you're using a self-managed deployment, restart {kib}.
+
+[discrete]
+[[obs-ai-interact]]
+= Interact with the Observability AI Assistant
+
+You can find Observability AI Assistant prompts throughout {observability} that provide the following information:
+
+- *Universal Profiling* – explains the most expensive processes and functions in your fleet and provides optimization suggestions.
```
Comment (SeanHeelan) [inline docs/en/observability/observability-ai-assistant.asciidoc:86]:
```suggestion
- *Universal Profiling* – explains the most expensive libraries and functions in your fleet and provides optimization suggestions.
```

**Review by SeanHeelan — COMMENTED** [review]
Nice work Mike! I've left a few comments and questions.

File: docs/en/observability/observability-ai-assistant.asciidoc (hunk 5 above)
Comment (dgieselaar) [reply docs/en/observability/observability-ai-assistant.asciidoc:6]: yes, definitely

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+- *Application performance monitoring (APM)* – explains APM errors and provides remediation suggestions.
+- *Infrastructure Observability* – explains host process and provides suggestions for optimizing CPU usage.
```
[the remark this answers is not in the document]
Comment (roshan-elastic) [reply docs/en/observability/observability-ai-assistant.asciidoc:83]: All good @SeanHeelan cheers

File: docs/en/observability/observability-ai-assistant.asciidoc (hunk 6 above)
Comment (mdbirnstiehl) [reply docs/en/observability/observability-ai-assistant.asciidoc:17]: Sounds good, @SeanHeelan. I think this is the last thing we need unless others want to review. I'll wait on this update to merge.

**Comment by bmorelli25** [comment]
@mdbirnstiehl can we merge this?

**Pull request merged**